src/context/QRContext.tsx

```
@@ -35,76 +44,180 @@ const getSafeLocalStorage = () => {
+    registerSignal: (name, callback) => {
+      signals[name].add(callback);
+      return () => signals[name].delete(callback);
+    }
+  };
+
+  return store;
+};
 
+export const QRProvider = ({ children, initialConfig }: { children: React.ReactNode, initialConfig?: Partial<QRConfig> }) => {
+  const storeRef = useRef<QRStore | undefined>(undefined);
+  if (!storeRef.current) {
```

ESLint

[error] Error: Cannot access refs during render

React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef).

/home/jailuser/git/src/context/QRContext.tsx:166:8
  164 | export const QRProvider = ({ children, initialConfig }: { children: React.ReactNode, initialConfig?: Partial<QRConfig> }) => {
  165 |   const storeRef = useRef<QRStore | undefined>(undefined);
> 166 |   if (!storeRef.current) {
      |        ^^^^^^^^^^^^^^^^ Cannot access ref value during render
  167 |     storeRef.current = createQRStore(initialConfig);
  168 |   }
  169 |   

To initialize a ref only once, check that the ref is null with the pattern `if (ref.current == null) { ref.current = ... }`

(react-hooks/refs)